BaseStation/MainWindow/MapDisplay.xaml.cs

```diff
@@ -31,24 +32,37 @@ private void DisplayMap()
             // load in individual images
             // var path = Path.Combine(System.Environment.CurrentDirectory, "Images", "map.jpg");
             // TODO get the path from the settings
-            var uri = new Uri("Images/map.jpg", UriKind.Relative);
-            var bitmap = new BitmapImage(uri);
-            var hi = new BitmapImage(uri);
-            var image = new Image { Source = bitmap, Width = 300, Height = 300 };
-            var image2 = new Image { Source = bitmap, Width = 300, Height = 300 };
-            // put images to together 
-            Canvas.SetLeft(image, 0);
-            Canvas.SetTop(image, 0);
-            Canvas.SetLeft(image2, 300);
-            Canvas.SetTop(image2, 300);
-            canvas.Children.Add(image);
-            canvas.Children.Add(image2);
-            allImages[0] = image;
-            allImages[1] = image2;
+            MapTileDownloadManager.Configuration config = new MapTileDownloadManager.Configuration
+                (new Tuple<double, double>(47.653683, -122.304836), new Tuple<int, int>(300, 300), 19);
+            //MapTileDownloadManager.DownloadNewTileSet(new Tuple<int, int>(20,20), config, "test");
+            // map tile set is already downloaded. But this is a good example line
+            using (StreamReader file = new StreamReader(Directory.GetCurrentDirectory() + @"\Images\test.evm"))
+            {
+                int index = 0;
+                //first line specifies some config of map(not needed for now)
+                string line = file.ReadLine();
+                while ((line = file.ReadLine()) != null)
+                {
+                    string[] parts = line.Split('|');
+                    string[] location = parts[0].Split(',');
+                    var uri = new Uri(Directory.GetCurrentDirectory() + @"\Images\" + parts[1] + ".jpg", UriKind.Absolute);
```

Comment:
Extract adding a new image to a function, maybe called `AddTile(String, int, int)`